src/Microsoft.Diagnostics.Monitoring.EventPipe/EventSourcePipeline.cs

```diff
@@ -86,7 +86,19 @@ public async Task<Task> StartAsync(CancellationToken token)
             // started task. Logically, the run task will not successfully complete before the session
             // started task. Thus, the combined task completes either when the session started task is
             // completed OR the run task has cancelled/failed.
-            await Task.WhenAny(_processor.Value.SessionStarted, runTask).Unwrap().ConfigureAwait(false);
+            try
+            {
+                await Task.WhenAny(_processor.Value.SessionStarted, runTask).Unwrap().ConfigureAwait(false);
+            }
+            catch (TaskCanceledException)
+            {
+                if (runTask.IsFaulted)
+                {
+                    throw runTask.Exception.InnerException;
```

Comment:
What is the benefit of throwing this exception instead of throwing the TaskCanceledException?
Separately, does anything guarantee that runTask.Exception.InnerException is non-null? Not all exceptions have an inner exception.